File: webapi/Models/Product.cs
namespace webapi.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public int Price { get; set; }
        public string[] Tags { get; set; } = new string[0];
        public bool Favorite { get; set; } = false;
        public string ImgUrl { get; set; } = string.Empty;
    }
}

File: angularapp/src/app/models/Product.ts
/**
 * Represents a product in the online store.
 */
export class Product {

  // The name of the product.
  name!: string;

  // The price of the product. 
  price!: number;

  // Whether or not the product is a favorite.
  favorite: boolean = false;

  // The URL for the product's image.
  imgUrl!: string;

  stock!: number;

}
